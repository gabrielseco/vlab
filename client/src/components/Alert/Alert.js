import React from 'react';
import PropTypes from 'prop-types';
import applyClasses from 'apply-classes';
import styles from './Alert.scss';

const Alert = ({ children, danger }) => {
  const classes = applyClasses({
    [styles.alert]: true,
    [styles.danger]: danger
  });
  return <div className={classes}>{children}</div>;
};

Alert.propTypes = {
  children: PropTypes.any,
  danger: PropTypes.bool
};

export default Alert;
