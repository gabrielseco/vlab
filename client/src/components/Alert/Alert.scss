.alert {
  display: block;
  padding: 10px;
  width: 100%;
}

.danger {
  background: #ff0000;
  color: #ffffff;
}
