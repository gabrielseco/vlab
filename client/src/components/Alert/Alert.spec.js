import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Alert from './Alert';

describe('Alert suite', () => {
  const component = shallow(<Alert danger>Mensaje de alerta</Alert>);

  it('should render the Alert component', () => {
    expect(component).toBeDefined();
  });

  it('should create the snapshot with props children and danger', () => {
    const tree = renderer.create(component).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
