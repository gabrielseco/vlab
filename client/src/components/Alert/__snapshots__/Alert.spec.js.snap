// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert suite should create the snapshot with props children and danger 1`] = `
<div
  className="alert danger"
>
  Mensaje de alerta
</div>
`;
